### PR TITLE
Dialyzer Fixes

### PR DESCRIPTION
**Pull request by asummers**
Resolved about 10 dialyzer errors, and converted case with true/false to if/else. 

Discussion: the event handlers seem to all be doing the same high level logic. Perhaps it might be better to expose a `validator/0` that returns a validator module, and move the with/else logic to the caller. Haven't gone through all of them yet, but I could see that being advantageous.

### CODE REVIEW

File: lib/code_corps/github/event/pull_request/pull_request.ex [hunk 1]
```diff
@@ -33,12 +32,12 @@ defmodule CodeCorps.GitHub.Event.PullRequest do
     end
   end
 
-  @spec validate_payload(map) :: {:ok, :valid}
-                               | {:error, :unexpected_payload}
+  @spec validate_payload(map) :: {:ok, :valid} | {:error, :unexpected_payload}
```
Comment (joshsmith) [inline lib/code_corps/github/event/pull_request/pull_request.ex:35]: I think the old spec is maybe how the formatter is doing it now?

**Review by joshsmith — COMMENTED** [review]
Overall looks good. One minor comment.

File: lib/code_corps/github/event/pull_request/pull_request.ex (hunk 1 above)
Comment (asummers) [reply lib/code_corps/github/event/pull_request/pull_request.ex:35]: Oh possibly. It was inconsistent across the handlers, so I just normalized them by hand. I can change it back if you want, but considering the discussion note in the PR, I feel like this code could be moved to the caller at some point soon. Your call, though.

**Comment by joshsmith** [comment]
Can you rebase and squash into a single commit? Then I'll approve and we can merge.

**Review by joshsmith — APPROVED** [review]

**Pull request merged**